### PR TITLE
Simple solution  to validate toml in issue #106104

### LINKED ISSUE #106104: Validate `triagebot.toml` in tidy/CI
Would've prevented #106102, which broke in #105661. 

We should at least validate the toml file is valid syntax, but ideally we'd validate more context-specific stuff.

### PR DESCRIPTION
**Pull request by aadityadhruv**
Fixes https://github.com/rust-lang/rust/issues/106104 using the `toml` crate to validate if the `triagebot.toml` file is in valid toml.

### CODE REVIEW

File: src/tools/tidy/src/triagebot.rs [hunk 1]
```diff
@@ -0,0 +1,24 @@
+use crate::walk::{filter_dirs, walk};
+use serde::Deserialize;
+use std::path::Path;
+use toml::Value;
+
+#[derive(Deserialize)]
+struct Config {}
```
Comment (bjorn3) [inline src/tools/tidy/src/triagebot.rs:7]: This seems to be unused. Can you remove it and remove the serde dependency and serde feature of the toml dependency?

File: src/tools/tidy/src/lib.rs [hunk 2]
```diff
@@ -64,6 +64,7 @@ pub mod pal;
 pub mod primitive_docs;
 pub mod style;
 pub mod target_specific_tests;
+pub mod triagebot;
```
Comment (bjorn3) [inline src/tools/tidy/src/lib.rs:67]: You will still need to call `triagebot::check` in main.rs.

File: src/tools/tidy/src/triagebot.rs [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+pub fn check(path: &Path, bad: &mut bool) {
+    walk(path, &mut |path| filter_dirs(path), &mut |entry, contents| {
+        let file = entry.path();
+        let filename = file.file_name().unwrap();
+        if filename != "triagebot.toml" {
```
Comment (bjorn3) [inline src/tools/tidy/src/triagebot.rs:13]: Maybe only check the top-level `triagebot.toml` instead of trying to find it in all subdirectories?

Comment (aadityadhruv) [reply src/tools/tidy/src/triagebot.rs:13]: Sure, I'll fix it. I had searched the repo for the `triagebot.toml` files and saw multiple, hence I thought maybe I should do a walk. Is there a reason we don't check those toml files?

Comment (bjorn3) [reply src/tools/tidy/src/triagebot.rs:13]: Only the top-level one is actually used by triagebot afaik. Several subdirectories of `src/tools` are submodules or subtrees. Their copy is only used for the separate repo from which they are pulled in. For example `src/tools/miri` is a subtree pulled from https://github.com/rust-lang/miri/, so the https://github.com/rust-lang/miri/ repo has a `triagebot.toml` file which is used when interacting with this repo, but it is also copied into `src/tools/miri/triagebot.toml` as part of `git subtree` syncs, as everything is copied.

**Comment by jyn514** [comment]
Sorry, I meant to reply on the issue. I liked Eric's idea of validating this in triagebot, I don't think it should go in tidy. Fortunately I think nearly all of the code you've written can stay the same :) you'd just need to handle posting a comment on the PR if the check fails.

**Comment by aadityadhruv** [comment]
Got it, I'll get to work on that! Thanks!

**Pull request closed without merging**